### PR TITLE
Fix additional Chef 12.x migration bugs, upgrade to Kafka 11

### PR DESCRIPTION
**Pull request by http-418**
This is the most recent set of changes I've stumbled across on my hardware cluster builds.

- Default to Ubuntu 14.04 + legacy BIOS to improve VM build experience
- Update to Kafka 0.11 from Apache.org
- Force restarts of the bootstrap xinetd on every chef run
- Remove make_databag.sh and friends
- Fix bach_repository::gems permissions
- Fix bad knife.rb using a Chef 12.x client

### CODE REVIEW

**Comment by http-418** [comment]
This is blocked on fixing #890

**Comment by cbaenziger** [comment]
Can pull request #890's code be pulled out of this to get it moving please?

File: cookbooks/bcpc/recipes/cobbler.rb [hunk 1]
```diff
@@ -345,3 +350,13 @@
   action [:enable,:start]
   notifies :run, 'bash[cobbler-sync]', :delayed
 end
+
+#
+# After recurring problems with reloading the xinetd
+# configuration, we resort to a forced restart on every chef run.
```
Comment (cbaenziger) [inline cookbooks/bcpc/recipes/cobbler.rb]: Do you have some concrete pointers which can be referenced so we know when this may be pulled out?

**Comment by http-418** [comment]
This is partly tested.  It's running on my (hardware) lab cluster right now.  I would like to see a VM build complete, though.

File: cookbooks/bcpc/recipes/cobbler.rb (hunk 1 above)
Comment (http-418) [inline cookbooks/bcpc/recipes/cobbler.rb]: I still don't think this commit has fixed my problem.  xinetd fails to respond on the TFTP port when the VMs boot, and I don't know why.  Worse, it works again after another chef run.

**Comment by ekund** [comment]
Setting oozie URL to the management vip doesn't work.  I believe you need to set it to the address of the node that actually runs oozie on it.  As it's not HA, it's just on one node.
Also, the url should be of the form: <url>:<port>/oozie so it won't work in any case.  Can you please fix?

**Pull request merged**